internal/services/search/search_service_data_source.go

```diff
@@ -116,6 +121,10 @@ func dataSourceSearchServiceRead(d *pluginsdk.ResourceData, meta interface{}) er
 			replicaCount := 1
 			publicNetworkAccess := true
 
+			if props.EncryptionWithCmk != nil && props.EncryptionWithCmk.EncryptionComplianceStatus != nil {
```

Comment:
```suggestion
			if props.EncryptionWithCmk != nil && props.EncryptionWithCmk {
```

Reply:
props.EncryptionWithCmk != nil && props.EncryptionWithCmk wont compile, so I will remove the nil check for the subproperty as it will be done in pointer.From